text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,678 @@
+  - By way of example, a target that runs within a sandbox may need to modify
+    the handling of files, tool invocation, and similar to meet the
+    expectations and conventions of the sandbox, but must not introduce a
+    separate "sandboxed compilation" interface separate from the CLI interface
+    without going through the normal approval process for such an interface.
+    Such an interface should take into account potential other targets with
+    similar sandboxes.
+- If the host tools for the platform would normally be expected to be signed or
+  equivalent (e.g. if running unsigned binaries or similar involves a
+  "developer mode" or an additional prompt), it must be possible for the Rust
+  project's automated builds to apply the appropriate signature process,
+  without any manual intervention by either Rust developers, target
+  maintainers, or a third party.  This process must meet the approval of the
+  infrastructure team.
+  - This process may require one-time or semi-regular manual steps by the
+    infrastructure team, such as registration or renewal of a signing key. Any
+    such manual process must meet the approval of the infrastructure team.
+  - This requirement exists to ensure that Rust builds, including nightly
+    builds, can meet the necessary requirements to allow users to smoothly run
+    the host tools.
+- Providing host tools does not exempt a target from requirements to support
+  cross-compilation if at all possible.
+- All requirements for tier 2 apply.
+
+A target may be promoted directly from tier 3 to tier 2 with host tools if it
+meets all the necessary requirements, but doing so may introduce substantial
+additional complexity. If in doubt, the target should qualify for tier 2
+without host tools first.
+
+## Tier 1 target policy
+
+At this tier, the Rust project guarantees that a target builds and passes all
+tests, and will reject patches that fail to build or pass the testsuite on a
+target. We hold tier 1 targets to our highest standard of requirements.
+
+A proposed new tier 1 target must be reviewed and approved by the compiler team
+based on these requirements. In addition, the release team must approve the
+long-term viability of the target, and the additional work of supporting the
+target. For a tier 1 target, this will typically take place via a full RFC
+proposing the target, to be jointly reviewed and approved by the compiler team
+and release team.
+
+In addition, the infrastructure team must approve the integration of the target
+into Continuous Integration (CI), and the tier 1 CI-related requirements. This
+review and approval may take place in a PR adding the target to CI, by an
+infrastructure team member reporting the outcome of a team discussion, or by
+including the infrastructure team in the RFC proposing the target.
+
+- Tier 1 targets must have substantial, widespread interest within the
+  developer community, and must serve the ongoing needs of multiple production
```

Comment:
I would expect a tier 1 target to have a substantial number of people and groups eagerly working with the tier 2 support, and then a larger group eagerly waiting for tier 1. I also think "we're ready to put this into production" or "we're porting to this target" or "we're planning to use this target in production" will all qualify; it's not a hard bar to meet, if people are actually using the target. The qualifier "production" here is intended to disqualify targets used exclusively as toys, hobbies, "because we can", and similar (such targets would top out at tier 2); the qualifier "ongoing needs" is intended to disqualify targets that were *once* production but are no longer used for any production purpose. I think a target aiming for tier 1 will have little trouble demonstrating sufficient interest.

> Put another way, if I want to use Rust on a platform and have it supported for production use, how would my organization go about breaking that dependency cycle?

Get it added as tier 3, work up to tier 2, get more projects or organizations using it, demonstrate *widespread* need for tier 1.